src/components/List.js

```diff
@@ -79,6 +80,39 @@ const List = ({ token }) => {
     return daysUntilNextPurchase > 0 ? daysUntilNextPurchase : 0;
   };
 
+  const sortedList = docs.sort((a, b) => {
+    const aDaysUntilNextPurchase = getDaysUntilNextPurchase(a);
+    const bDaysUntilNextPurchase = getDaysUntilNextPurchase(b);
+
+    if (aDaysUntilNextPurchase < bDaysUntilNextPurchase) {
+      return -1;
+    }
+    if (aDaysUntilNextPurchase > bDaysUntilNextPurchase) {
+      return 1;
+    }
+    if (aDaysUntilNextPurchase === bDaysUntilNextPurchase) {
+      if (a.itemName < b.itemName) {
+        return -1;
+      } else {
+        return 1;
+      }
+    }
+    return 0;
+  });
+
```

Comment:
Wow! I am super impressed. The code for the `sortedList` and `backgroundColor` code blocks is slick! There's thing I'm wondering about - I might be misunderstanding the AC, but I think there may be an issue with whether or not the `sortedList` is getting re-sorted as the `daysUntilNextPurchase` value changes.

* Initially we use the radio button value of 7, 14 or 30 for our `timeFrame`.
* Then once the user begins to purchase the item, the `timeFrame` is based on `daysUntilNextPurchase` instead of the 7, 14 and 30 value <i>and</i> our estimate.js calculation gets "smarter".
* Question :  do our background colors change accordingly as the app gets smarter?

Here is a snapshot of the list `stroll ames knack` which might be part of my confusion : 
<img width="237" alt="background-colors" src="https://user-images.githubusercontent.com/36529777/107451956-31d14300-6afd-11eb-8b2b-99b9e4f5e557.PNG">

If the background colors should be similar `according to how soon the item is expected to be bought again` <i>and</i> `items should be sorted by the estimated number of days until next purchase`, should items with similar backgrounds be grouped together and have similar estimated `time until next purchase`?

Is this possibly a situation where we would need something like a `useEffect` hook or some other form of state to reset the value of `sortedList` any time the docs are updated?

Reply:
> Question : Are our background colors supposed to change accordingly as the app gets smarter?

Good question. We're using the `getDaysUntilNextPurchase` function to inside of the `backgroundColor` function. We think the colors should update properly since `getDaysUntilNextPurchase` uses `timeFrame` which now gets the return from `calculateEstimate`

> should items with similar backgrounds be grouped together and have similar estimated time until next purchase? I get the impression that this might be the case due to Stacie's demo around second 0:25

We were unsure of how to handle the sorting of `inactive` items on the list. I see what you're saying about the demo and how all inactive items are at the end of the list. At the moment active and inactive items are grouped together by time frame. This is definitely a situation where we are looking for input from the team. 😄 

> Is this possibly a situation where we would need something like a useEffect hook or some other form of state to reset the value of sortedList any time the docs are updated?

We considered using useEffect but wasn't sure if it was necessary.  Also another place where we are looking for input from the team. 😄

Reply:
Currently testing it out with netlify preview, the sorting is really cool, it definitely has a smart feel to it as you check an item and the list is reordered. 
I get easily confused so it took me a while to wrap my head around the difference in the sorting and the background color lol. The items are sorted based on the number of days until the next purchase, and the background color is just to represent how soon a user initially expected to need the item again, right?

Reply:
The latest code I pushed up now includes sorting of the inactive items.